app/components/earthworks/code_snippet_modal_component.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module Earthworks
+  class CodeSnippetModalComponent < ViewComponent::Base
+    def initialize(document)
+      @document = document
+      @raster_data_type = raster_data?
+      @vector_data_type = vector_data?
+      super
+    end
+
+    def code_block(content)
+      sanitize "<pre><code>#{h content}</code></pre>"
+    end
+
+    def render_python
+      substitute_values(retrieve_text('Python'))
+    end
+
+    def render_r
+      substitute_values(retrieve_text('R'))
+    end
+
+    def render_leaflet
+      substitute_values(retrieve_text('Leaflet'))
+    end
+
+    # Incorporate
+    def substitute_values(content)
+      sub_content = content
+
+      # replace DRUID_ID
+      sub_content = sub_content.gsub('<<WXS_ID>>', @document.wxs_identifier) unless @document.wxs_identifier.empty?
+
+      # If layer id is required
+      sub_content = sub_content.gsub('<<LAYER_ID>>', layer_id) unless layer_id.nil?
+
+      # replace WMS reference
+      unless @document.references.wms.nil? || @document.references.wms.endpoint.nil?
+        sub_content = sub_content.gsub('<<GEOSERVER_WMS>>', @document.references.wms.endpoint)
+      end
+
+      # replace WFS reference
+      unless @document.references.wfs.nil? || @document.references.wfs.endpoint.nil?
+        sub_content = sub_content.gsub('<<GEOSERVER_WFS>>', @document.references.wfs.endpoint)
+      end
+
+      # Replace Bounding box information
+      unless @document.geom_field.empty?
+        geometry = @document.geometry.geom
+        bbox_array = geometry.split('ENVELOPE(')[1].split(')')[0].split(',').map(&:strip)
```

Comment:
In some cases, documents have no geometry (although after a full reindex after https://github.com/sul-dlss/searchworks_traject_indexer/pull/1505 takes effect, I believe this case will no longer exist.) 
But for now, if a document has no geometry, I am getting this error when trying to parse the ENVELOPE:
```
undefined method `split' for nil

bbox_array = geometry.split('ENVELOPE(')[1].split(')')[0].split(',').map(&:strip)
```

Reply:
I tried to handle this by adding another check for @ document.geometry.geom.